src/analyses/base.ml

```diff
@@ -1249,7 +1249,7 @@ struct
           (* If there's a non-heap var or an offset in the lval set, we answer with bottom *)
           (* If we're asking for the BlobSize from the base address, then don't check for offsets => we want to avoid getting bot *)
           if AD.exists (function
-              | Addr (v,o) -> (not @@ ctx.ask (Queries.IsHeapVar v)) || (if not from_base_addr then o <> `NoOffset else false)
+              | Addr (v,o) -> (not @@ ctx.ask (Queries.IsDynamicallyAlloced v)) || (ctx.ask (Queries.IsDynamicallyAlloced v) && not @@ ctx.ask (Queries.IsHeapVar v)) || (if not from_base_addr then o <> `NoOffset else false)
```

Comment:
Should use parenthesis instead of `@@`s here because it's not obvious where the `@@` ends.
Also, the query is duplicated, so it would be nice to just use a variable for it and make this line more readable.

Reply:
Should be fixed in https://github.com/goblint/analyzer/pull/1179/commits/3bdc92d7c6e58520f6ed552cb687fa158adcf429